aot/src/main/java/com/dylibso/chicory/aot/AotEmitters.java

```diff
@@ -0,0 +1,147 @@
+        var localType = AotUtil.localType(ctx.getType(), ctx.getBody(), loadIndex);
+        int opcode;
+        switch (localType) {
+            case I32:
+                opcode = Opcodes.ILOAD;
+                break;
+            case I64:
+                opcode = Opcodes.LLOAD;
+                break;
+            case F32:
+                opcode = Opcodes.FLOAD;
+                break;
+            case F64:
+                opcode = Opcodes.DLOAD;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported load target type: " + localType);
+        }
+        asm.visitVarInsn(opcode, loadIndex + 1); // +1 because local 0 is 'this'.
+    }
+
+    public static void LOCAL_SET(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        var storeIndex = (int) ins.operands()[0];
+        var localType = AotUtil.localType(ctx.getType(), ctx.getBody(), storeIndex);
+        int opcode;
+        switch (localType) {
+            case I32:
+                opcode = Opcodes.ISTORE;
+                break;
+            case I64:
+                opcode = Opcodes.LSTORE;
+                break;
+            case F32:
+                opcode = Opcodes.FSTORE;
+                break;
+            case F64:
+                opcode = Opcodes.DSTORE;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported store target type: " + localType);
+        }
+        asm.visitVarInsn(opcode, storeIndex + 1); // +1 because local 0 is 'this'.
+    }
+
+    public static void I32_ADD(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IADD);
+    }
+
+    public static void I32_AND(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IAND);
+    }
+
+    public static void I32_MUL(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IMUL);
+    }
+
+    public static void I32_OR(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IOR);
+    }
+
+    public static void I32_REM_S(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IREM);
+    }
+
+    public static void I32_SHL(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.ISHL);
+    }
+
+    public static void I32_SHR_S(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.ISHR);
+    }
+
+    public static void I32_SHR_U(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IUSHR);
+    }
+
+    public static void I32_SUB(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.ISUB);
+    }
+
+    public static void I32_XOR(AotContext ctx, Instruction ins, MethodVisitor asm) {
+        asm.visitInsn(Opcodes.IXOR);
+    }
+
+    public static BytecodeEmitter intrinsify(OpCode opcode, Class<?> staticHelpers) {
```

Comment:
I think you explained it well in the meeting this week, but could we get some inline docs about what inftinsifying is in this context? any added docs on some of the meatier pieces will be helpful as people start to contribute or go to fix bugs.

Reply:
I had somewhat documented these concepts in a couple other scattered areas, but added some documentation here as well.